### PR TITLE
Add methods for 'search_settings' endpoint

### PR DESCRIPTION
**Pull request by JanmanX**

### CODE REVIEW

File: elastic_app_search/client.py [hunk 1]
```diff
@@ -289,6 +289,46 @@ def click(self, engine_name, options):
         endpoint = "engines/{}/click".format(engine_name)
         return self.session.request_ignore_response('post', endpoint, json=options)
 
+    def show_settings(self, engine_name):
```
Comment (JasonStoltz) [inline elastic_app_search/client.py]: can you rename this to `get_search_settings` to stay consistent with the API Spec in the php client? https://github.com/elastic/app-search-php/blob/master/resources/api/api-spec.yml#L471

```suggestion
    def get_search_settings(self, engine_name):
```

File: elastic_app_search/client.py [hunk 2]
```diff
@@ -289,6 +289,46 @@ def click(self, engine_name, options):
...
+        Show all Weights and Boosts applied to the search fields of an Engine.
+        See https://swiftype.com/documentation/app-search/ for more details
+        on options and return values.
+
+        :param engine_name: Name of engine to search over.
+        """
+        endpoint = "engines/{}/search_settings".format(engine_name)
+        return self.session.request('get', endpoint)
+
+    def update_settings(self, engine_name, search_fields, result_fields, boosts):
```
Comment (JasonStoltz) [inline elastic_app_search/client.py]: can you rename this to `update_search_settings` for the same reason

```suggestion
    def update_search_settings(self, engine_name, search_fields, result_fields, boosts):
```

File: elastic_app_search/client.py [hunk 3]
```diff
@@ -289,6 +289,46 @@ def click(self, engine_name, options):
...
+        """
+        endpoint = "engines/{}/search_settings".format(engine_name)
+        options = {
+            'search_fields': search_fields,
+            'result_fields': result_fields,
+            'boosts':        boosts
+        }
+        return self.session.request('put', endpoint, json = options)
+
+    def reset_settings(self, engine_name):
```
Comment (JasonStoltz) [inline elastic_app_search/client.py]: can you rename this to `reset_search_settings` for the same reason
```suggestion
    def reset_search_settings(self, engine_name):
```

File: elastic_app_search/client.py [hunk 4]
```diff
@@ -289,6 +289,46 @@ def click(self, engine_name, options):
         endpoint = "engines/{}/click".format(engine_name)
         return self.session.request_ignore_response('post', endpoint, json=options)
 
+    def show_settings(self, engine_name):
+        """
+        Show all Weights and Boosts applied to the search fields of an Engine.
+        See https://swiftype.com/documentation/app-search/ for more details
+        on options and return values.
+
+        :param engine_name: Name of engine to search over.
```
Comment (JasonStoltz) [inline elastic_app_search/client.py]:
```suggestion
        :param engine_name: Name of engine.
```

File: elastic_app_search/client.py [hunk 5]
```diff
@@ -289,6 +289,46 @@ def click(self, engine_name, options):
...
+        endpoint = "engines/{}/search_settings".format(engine_name)
+        return self.session.request('get', endpoint)
+
+    def update_settings(self, engine_name, search_fields, result_fields, boosts):
+        """
+        Updates the settings for Search Field (Weights), Result Field, and Boosts.
+        See https://swiftype.com/documentation/app-search/ for more details
+        on options and return values.
+
+        :param engine_name: Name of engine to search over.
```
Comment (JasonStoltz) [inline elastic_app_search/client.py]:
```suggestion
        :param engine_name: Name of engine.
```

File: elastic_app_search/client.py [hunk 6]
```diff
@@ -289,6 +289,46 @@ def click(self, engine_name, options):
...
+        }
+        return self.session.request('put', endpoint, json = options)
+
+    def reset_settings(self, engine_name):
+        """
+        Reset your Engine to default values.
+        See https://swiftype.com/documentation/app-search/ for more details
+        on options and return values.
+
+        :param engine_name: Name of engine to search over.
```
Comment (JasonStoltz) [inline elastic_app_search/client.py]:
```suggestion
        :param engine_name: Name of engine.
```

File: elastic_app_search/client.py [hunk 7]
```diff
@@ -289,6 +289,46 @@ def click(self, engine_name, options):
         endpoint = "engines/{}/click".format(engine_name)
```
Comment (JasonStoltz) [inline elastic_app_search/client.py]: Please also sign the CLA

Comment (JanmanX) [reply elastic_app_search/client.py]: I have applied your suggestions, but I wont be able to sign the CLA within the next few weeks.
I hope this PR can stay up, so I can merge it by then.

**Pull request closed without merging**